### PR TITLE
Customise field

### PR DESCRIPTION
**Pull request by davidyell**
I am upgrading a Cake2 application which used the CakeDC/Utils plugin SoftDelete behaviour, which means that my fields are not named correctly. Rather than update a field in every table of my application, I've just changed the code to allow the field to be declared in the Table.

### CODE REVIEW

**Comment by PGBI** [comment]
Thanks for this new feature. Could you just add a unit test for it?

**Comment by davidyell** [comment]
Sure thing, can do that tomorrow.

File: src/Model/Table/SoftDeleteTrait.php [hunk 1]
```diff
@@ -7,6 +7,20 @@
 trait SoftDeleteTrait {
 
     /**
+     * Get the configured deletion field
+     *
+     * @return string
+     */
+    protected function getField()
```
Comment (PGBI) [inline src/Model/Table/SoftDeleteTrait.php]: Isn't `getField` a too common name for a Table method? If a table using softDeleteTrait already has a `getField()` method, soft deleting would then fail.
I would use a less common name, like `getSoftDeleteField()`. What do you think?

File: README.md [hunk 2]
```diff
@@ -42,7 +42,20 @@ class UsersTable extends Table
...
+```php
+// in src/Model/Table/UsersTable.php
+...
+use SoftDelete\Model\Table\SoftDeleteTrait;
+
+class UsersTable extends Table
+{
+    use SoftDeleteTrait;
+    
+    public $softDeleteField = 'deleted_date';
```
Comment (PGBI) [inline README.md]: I think it's better if $softDeleteField is private.

File: tests/Fixture/TagsFixture.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+use Cake\ORM\Table;
+use Cake\TestSuite\Fixture\TestFixture;
+
+use SoftDelete\Model\Table\SoftDeleteTrait;
+
+class TagsTable extends Table
+{
+    use SoftDeleteTrait;
+
+    public $softDeleteField = 'deleted_date';
```
Comment (PGBI) [inline tests/Fixture/TagsFixture.php]: same remark as before: can you make this attribute private?

**Comment by PGBI** [comment]
Sorry for these late remarks (private attribute + change method name). These will be the last ones :)

File: src/Model/Table/SoftDeleteTrait.php (hunk 1 above)
Comment (davidyell) [inline src/Model/Table/SoftDeleteTrait.php]: Ah yes, I had not thought of that, it's a good point.

File: tests/Fixture/TagsFixture.php (hunk 3 above)
Comment (davidyell) [inline tests/Fixture/TagsFixture.php]: I don't think making it private in the Table class is wise. User land code may well extend the table class. It also gives people the ability to change the field on the fly by updating the table property from other places in their application.

Comment (PGBI) [inline tests/Fixture/TagsFixture.php]: okay let's go protected then!

**Comment by davidyell** [comment]
Thanks for merging the changes, are you okay to update the method name please? Sorry I didn't get to it, I was interested in your thoughts on the class property visibility.

**Comment by PGBI** [comment]
I've updated the method name already. I'm going to vacation, i wanted to merge your pr before leaving !

**Pull request merged**